cpp/src/io/orc/stripe_init.cu

```diff
@@ -239,11 +259,14 @@ static uint32_t __device__ ProtobufParseRowIndexEntry(rowindex_state_s* s,
                                                       uint8_t const* const end)
 {
   constexpr uint32_t pb_rowindexentry_id = ProtofType::FIXEDLEN + 8;
+  auto const stream_order                = index_order_from_index_types(s->chunk.skip_count);
 
   const uint8_t* cur      = start;
   row_entry_state_e state = NOT_FOUND;
-  uint32_t length = 0, strm_idx_id = s->chunk.skip_count >> 8, idx_id = 1, ci_id = CI_PRESENT,
-           pos_end = 0;
+  uint32_t length         = 0;
+  uint32_t idx_id         = 0;
+  uint32_t pos_end        = 0;
+  uint32_t ci_id          = CI_NUM_STREAMS;
```

Comment:
Hmm why did some of these values change?

Reply:
Great question!
`idx_id` was one because of the... complicated... way `strm_idx_id` was implemented. New implementation counts from zero (as expected).
`ci_id` is now `CI_NUM_STREAMS` to denote an invalid value. `CI_PRESENT` is a valid stream type in this context so I changed this to a value that makes sure the execution flow is as expected. `ci_id` should be set to the actual stream type before it's used. Unfortunately, we do need to keep track of the stream type outside of the loop.

Reply:
Yes, if the input is valid and the code is correct, line 316 will be executed before `ci_id` is used. If that's not the case, I'd rather it have an invalid value than an incorrect one :)